test/HighTable.test.tsx

```diff
@@ -157,3 +164,416 @@ describe('When sorted, HighTable', () => {
+    })
+    expect(onSelectionChange).toHaveBeenCalledWith({ ranges: [{ start, end: start + 1 }], anchor: start })
+    expect(queryByRole('row', { selected: true })?.getAttribute('aria-rowindex')).toBe(`${start + 2}`)
+  })
+
+  it('on data change, onSelection is called with an empty selection and the DOM is updated to unselect the rows', async () => {
+    const start = 2
+    const onSelectionChange = vi.fn()
+    const { rerender, findByRole, queryByRole } = render(<HighTable data={data} onSelectionChange={onSelectionChange}/>)
+    // await because we have to wait for the data to be fetched first
+    const cell = await findByRole('cell', { name: 'row 2' })
+    expect(onSelectionChange).not.toHaveBeenCalled()
+    onSelectionChange.mockClear()
+
+    // select a row
+    const rowHeader = cell.closest('[role="row"]')?.querySelector('[role="rowheader"]')
+    expect(rowHeader).not.toBeNull()
+    await act(async () => {
+      rowHeader && await userEvent.click(rowHeader)
+    })
+    expect(onSelectionChange).toHaveBeenCalledWith({ ranges: [{ start, end: start + 1 }], anchor: start })
+
+    rerender(<HighTable data={otherData} onSelectionChange={onSelectionChange}/>)
+    // await again, since we have to wait for the new data to be fetched
+    const other = await findByRole('cell', { name: 'other 2' })
+    expect(queryByRole('cell', { name: 'row 2' })).toBeNull()
+    expect(queryByRole('row', { selected: true })).toBeNull()
+    expect(onSelectionChange).toHaveBeenCalledWith({ ranges: [] })
+  })
+
+  it('passing the selection prop is ignored and a warning is printed in the console', async () => {
+    const start = 2
+    const selection = undefined
+    const onSelectionChange = vi.fn()
+    console.warn = vi.fn()
+
+    const { queryByRole, findByRole, rerender } = render(<HighTable data={data} selection={selection} onSelectionChange={onSelectionChange}/>)
+    // await because we have to wait for the data to be fetched first
+    await findByRole('cell', { name: 'row 2' })
+    expect(queryByRole('row', { selected: true })).toBeNull()
+    expect(console.warn).not.toHaveBeenCalled()
+
+    const newSelection = { ranges: [{ start, end: start + 1 }], anchor: start }
+    rerender(<HighTable data={data} selection={newSelection} onSelectionChange={onSelectionChange}/>)
+    // no need to await because the data is already fetched
+    expect(queryByRole('row', { selected: true })).toBeNull()
+    expect(console.warn).toHaveBeenNthCalledWith(1, expect.stringMatching(/cannot be set to a value/))
+  })
+})
+
+describe('in disabled selection state (neither selection nor onSelection props), ', () => {
+  const data = {
+    header: ['ID', 'Count'],
+    numRows: 1000,
+    rows: (start: number, end: number) => Promise.resolve(
+      Array.from({ length: end - start }, (_, index) => ({
+        ID: 'row ' + (index + start),
+        Count: 1000 - start - index,
+      }))
+    ),
+  }
+
+  const otherData = {
```

Comment:
otherData is duplicated, can we just hoist it to the top?

Reply:
done (and `data` too)